x-pack/plugins/alerting/server/lib/get_security_health.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ILicenseState } from '../lib/license_state';
+import { EncryptedSavedObjectsPluginSetup } from '../../../encrypted_saved_objects/server';
+
+export interface SecurityHealth {
+  isSufficientlySecure: boolean;
+  hasPermanentEncryptionKey: boolean;
+}
+
+export const getSecurityHealth = async (
+  licenseState: ILicenseState | null,
+  encryptedSavedObjects: EncryptedSavedObjectsPluginSetup,
+  areApiKeysEnabled: () => Promise<boolean>
```

Comment:
It makes sense if I read where you extracted this function from. Without that context these three arguments feel slightly inconsistent in that

- license state is injected as a class that has a getter
- encryption capabilities are injected via a whole plugin contract that has a getter
- the api keys getter is injected individually

Just passing three getters would make the signature more predictable. This shows in the call site and the complexity of the test setup.

But I'm being super nit-picky and opinionated, so it's not worth delaying the PR based on this. :innocent:

Reply:
Yep, that's fair. In https://github.com/elastic/kibana/pull/117505/commits/489df362a0da02b5eaa4516a386342803070691a I've simplified the signature down to three getters (`isEsSecurityEnabled`, `isAbleToEncrypt`, and `areApiKeysEnabled`) and pushed the complexity out to the callers.